materials/section-2/README.md

```diff
@@ -0,0 +1,149 @@
+
+```clojure
+(ns <name>.handler
+  (:require [compojure.core :refer :all]
+            [compojure.route :as route]
+            [ring.middleware.defaults :refer [wrap-defaults site-defaults]]))
+
+(defroutes app-routes
+  (GET "/" [] "Hello World")
+  (route/not-found "Not Found"))
+
+(def app
+  (wrap-defaults app-routes site-defaults))
+```
+
+Small breakdown:
+
+```clojure
+(ns <name>.handler
+  (:require [compojure.core :refer :all]
+            [compojure.route :as route]
+            [ring.middleware.defaults :refer [wrap-defaults site-defaults]]))
+```
+
+Namespace is defined and the core of Compojure is included along with the route namespace. It also requires the middleware namespace from Ring. Middlewares are just an easy way to handle requests before they reach your route handler. Ring has a few default ones which Compojure uses when it bootstraps the project.
+
+```clojure
+(defroutes app-routes
+  (GET "/" [] "Hello World")
+  (route/not-found "Not Found"))
+```
+
+`defroutes` is a macro provided by Compojure. It takes two parameters. A name and route handlers. A route handler is nothing more than a basic function. Compojure provides a handful of route handlers. `GET`, `POST`, `PATCH`, `PUT`, `HEAD`, `DELETE`, `OPTIONS` and `ANY`.
+All of these route handlers work in the same way. Each of them take a path as the first argument, a request body, and a route body as the last argument.
+
+```clojure
+(GET "/home" request (str "Welcome Home " ((:params request) "person")  "!"))
+  ^     ^       ^                      ^
+method route  request                body
+```
+
+The request body is quite unique. Compojure will compile the request differently depending on what you have given it as the second argument. This way of defining routes can be a bit unclear, as *all request params are always strings and not keywords*, and fetching the params key from the request object can be very tiring and lead to verbose code. The most common practice is to pass it a vector. Compojure will automatically destruct the request object and pass the request parameters present in the vector into your route body.
+
+```clojure
+(GET "/home" [person] (str "Welcome Home " person "!"))
+  ^     ^       ^                      ^
+method route  params                body
+```
+
+This is the more common approach when writing routes.
+
+The last part:
+
+```clojure
+(def app
+  (wrap-defaults app-routes site-defaults))
+```
+
+This is the bootstrapped application. Ring will look for this _var_ when starting the app. The app _var_ can be wrapped in middleware layers. Compojure wraps our routes with some default middlewares provided by Ring.
+
+**These middlewares should be deleted!**
```

Comment:
Why? It might be useful to provide a clarification here

Reply:
This is actually I really good question! I tried to find the answer from ring's documentation. But according to the documentation it should work out of the box. I'm somewhat confused now